dotCMS/src/main/java/com/dotcms/publishing/PublisherAPIImpl.java

```diff
@@ -79,36 +79,32 @@ final public PublishStatus publish ( PublisherConfig config, PublishStatus statu
-                // will return true after that always.
-                final boolean bundleExists = BundlerUtil.bundleExists(config);
-
                 try (BundleOutput output = publisher.createBundleOutput();
-                     ManifestBuilder manifestBuilder = new CSVManifestBuilder()){
-
-                    config.setManifestBuilder(manifestBuilder);
-                    status.addOutput(output);
-                    // Run bundlers
-
-                    if (config.isStatic()) {
-                        //If static we just want to save the things that we need,
-                        // at this point only the id, static and operation.
-                        PublisherConfig pcClone = new PublisherConfig();
-                        pcClone.setId(config.getId());
-                        pcClone.setStatic(true);
-                        pcClone.setOperation(config.getOperation());
-                        Logger.info(this, "Writing bundle.xml file");
-                        BundlerUtil.writeBundleXML(pcClone, output);
-                    } else {
-                        Logger.info(this, "Writing bundle.xml file");
-                        BundlerUtil.writeBundleXML(config, output);
-                    }
+                     ManifestBuilder manifestBuilder = new CSVManifestBuilder()) {
+
+                    if (!output.bundleFileExists() || !publishAuditAPI.isPublishRetry(config.getId())) {
+                        output.create();
+                        config.setManifestBuilder(manifestBuilder);
+                        status.addOutput(output);
+                        // Run bundlers
+
+                        if (config.isStatic()) {
+                            //If static we just want to save the things that we need,
+                            // at this point only the id, static and operation.
+                            PublisherConfig pcClone = new PublisherConfig();
+                            pcClone.setId(config.getId());
+                            pcClone.setStatic(true);
+                            pcClone.setOperation(config.getOperation());
+                            Logger.info(this, "Writing bundle.xml file");
+                            BundlerUtil.writeBundleXML(pcClone, output);
+                        } else {
+                            Logger.info(this, "Writing bundle.xml file");
+                            BundlerUtil.writeBundleXML(config, output);
+                        }
 
-                    // If the bundle exists and we are retrying to push the bundle
-                    // there is no need to run all the bundlers again.
-                    if (!bundleExists || !publishAuditAPI.isPublishRetry(config.getId())) {
+                        // If the bundle exists and we are retrying to push the bundle
+                        // there is no need to run all the bundlers again.
+                        //if (!bundleExists || !publishAuditAPI.isPublishRetry(config.getId())) {
```

Comment:
this line can be removed

Reply:
done https://github.com/dotCMS/core/pull/20883/commits/597ac01ccceef14985485a785e5de9a4e0584623